[4294 : Resolved Tables overflow the content area if cell contains unbreakable text

hitesh-wagento opened the pull request:
## Purpose of this PR
Resolved Tables overflow the content area if cell contains unbreakable text

<!-- REQUIRED Describe the goal and the type of changes this PR covers. -->

This PR ...

See https://devdocs.magento.com/guides/v2.3/extension-dev-guide/webapi/request-processor-pool.html
Since all our tables have table-layout: auto issues like this sometimes can happen.



## Affected URLs
https://devdocs.magento.com/guides/v2.3/extension-dev-guide/webapi/request-processor-pool.html

<!-- REQUIRED List the URLs you are changing. Not needed for large numbers of files. -->


<!-- 
If you are fixing a Github issue, note it in the following format and the issue will automatically close when this PR is merged:
Fixes #4294

`master` is the default branch. PRs to this branch should be for current devdocs content. Merged PRs to master go live on the site automatically. Work for future releases generally goes in the `develop` branch. Work with the devdocs team if you are unsure where to submit your PR.
-->

devops-devdocs commented:
An admin must run tests on this PR before it can be merged.

jeff-matthews commented:
Related to #4294.

@hitesh-wagento, it looks like you added a new class to the local CSS file. Our styling primarily comes from the https://github.com/magento-devdocs/devdocs-theme project. There's already a class there that we should review (as mentioned in the bug report: `table-layout:fixed`. See https://github.com/magento-devdocs/devdocs-theme/pull/34 for more info and work with @belbiy on changes.

osrecio commented on common/css/devdocs.css:
```diff
@@ -1 +1 @@
-.home-contributors{margin-bottom:2em}.message-banner{margin:0}redoc[spec-url] table{table-layout:auto}redoc[spec-url] .menu-subitems .menu-item-title{text-transform:none}.language-bash.highlighter-rouge code:before{display:inline;content:'$';margin-right:5px;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.page-updated{margin:0 0 10px 0;color:#767676;cursor:he ...
+.home-contributors{margin-bottom:2em}.message-banner{margin:0}redoc[spec-url] table{table-layout:fixed}redoc[spec-url] .menu-subitems .menu-item-title{text-transform:none}.language-bash.highlighter-rouge code:before{display:inline;content:'$';margin-right:5px;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.page-updated{margin:0 0 10px 0;color:#767676;cursor:h ...
```
Maybe table should remain with `table-layout:auto`

belbiy replied:
Do not change this file directly, please. This file is generated from `scss/devdocs.scss`. To generate it, use `npm run build:css`

hitesh-wagento replied:
Hi @belbiy 

Thanks for your suggestion. I have one confusion in  https://github.com/magento-devdocs/devdocs-theme Project.

Do I need to fork clone this repo in https://github.com/hitesh-wagento/devdocs or in new place?

Thanks for advance

belbiy commented:
We already have a class name `fixed` that you can apply to tables. 
Here's how to use it:

```
| Cell 1 | Cell 2 | Cell 3 |
{: .fixed }
```

Please do not add class names to the `devdocs.css` directly as well, since we're using npm to generate that file from `scss/devdocs.scss`

hitesh-wagento commented on guides/v2.2/extension-dev-guide/webapi/request-processor-pool.md:
```diff
@@ -12,6 +12,7 @@ The request processors pool routes WebApi requests. It is located in the Magento
 
 Magento defines the following processors: 
 
+{:.webapitable}
```
Hi @belbiy 

I have added `.fixed` class in table as per your suggestion.

Thanks.

hitesh-wagento commented:
Hi @belbiy , @osrecio 

I have added `.fixed` class in table as per your suggestion and removed static css.

Thanks.

osrecio commented:
Hi @hitesh-wagento , I'm still seeing the css. Do you want that I delete it for you?

Thanks!

hitesh-wagento commented:
> Hi @hitesh-wagento , I'm still seeing the css. Do you want that I delete it for you?
> 
> Thanks!

Hi @osrecio ,

I have revert that css before I have changed and added css. if you need remove that CSS you can do it.

Thanks.
